docs/src/references/operation-workflow.md

```diff
@@ -0,0 +1,372 @@
+as well as the message payload attached to each state.
+
+However, there are some rules and best practices.
+
+- Three states are mandatory: `Init`, `Successful` and `Failed`.
+- `Init` must be the unique initial state of the workflow.
+  - This state must give all the information required to execute the command.
+  - Having no other initial states is important so any command requester (as the mapper)
+    can trigger commands in a systematic manner *even* if the workflow is updated.
+- `Successful` and `Failed` must be the unique terminal states of the workflow.
+  - The payload of the `Failed` state should contain a `reason` property telling what failed.
+  - Having no other terminal states is important so any command requester (as the mapper)
+    can await the termination of a command in a systematic manner *even* if the workflow is updated.
+- A workflow API should define *no-op* states with no pre-defined actions
+  and which sole purpose is to give an agent developer the opportunity to *insert* its own logic and extra steps.
+  - A *no-op* state is a state which has a single direct transition to the next state.
+  - As an example, having a `Downloaded` *no-op* state, with a direct transition to an `Install` state which uses the downloaded file,
+    let an agent operator override the automatic transition, say to check the downloaded file *before* moving the `Install` state.
+  - The `Init` state should be a *no-op* state.
+
+### Operation Implementation
+
+A workflow implementation for a specific operation must implement the actions specified for each non *no-op* state.
+This implementation has some freedom, notably to implement extra checks and actions but also to add new states.
+
+However, there are some rules and best practices.
+
+- All the state messages must be published as retained with QoS 1.
+- A workflow implementation should not reject a state message payload with unknown fields.
+  - It's also important to keep these unknown fields in the following states.
+  - This is important as we want to *extend* the workflow of an operation.
+    A software component added by the user might need these *extra* fields the plugin is not aware of.
+- A workflow implementation must not react on *no-op* states nor terminal states.
+  - The transition from a *no-op* state must be handled either by thin-edge as direct transition
+    or overridden by the user with domain-specific checks and actions.
+  - The terminal states, a.k.a `Successful` and `Failed`, are owned by the process which created the `Init` state (in practice, the mapper).
+    Only this process should clear the retained message state for an operation instance by sending an empty payload on command's topic.
+
+### Workflow Overriding
+
+Thin-edge provides a mechanism to override, extend and combine workflows.
+
+This mechanism is provided by the `tedge-agent` which gather a set of user-defined workflows
```

Comment:
Does  it mean that the `tedge-agent` is the owner of the full workflow?

Reply:
No. It means that `tedge-agent` manage the coordination and triggers the actions owned by tedge as running scripts and sending events.